common-libs/functions.sh

```diff
@@ -51,3 +51,38 @@ function enable_balance()
 		done
 	done
 }
+
+function create_file() {
+	tool=$1
+	run_mode=$2
+	local -n ref_required_dirs=$3
+    log_dir="/var/log/app"
+
+    # Check if directories exist, if not, create them
+    for dir in "${ref_required_dirs[@]}"; do
+        if [ ! -d "$log_dir/$dir" ]; then
+            mkdir -p "$log_dir/$dir"
+        fi
+    done
+
+    timestamp=$(date +%Y%m%d%H%M%S)
+
+    # Get the latest file number
+    last_file_number=$(ls $log_dir/"$tool"/"$run_mode" | grep $tool | grep $run_mode | sort -n | tail -n 1 | cut -c 1-1)
+
+    # If no files found create the first file
+    if [ -z "$last_file_number" ]; then
+        file_path="$log_dir/"$tool"/"$run_mode"/1_"$tool"_"$run_mode"-$timestamp.log"
+    else
+        # If files found, increment the last file number and create a new file
+        new_file_number=$((last_file_number + 1))
+        file_path="$log_dir/"$tool"/"$run_mode"/${new_file_number}_"$tool"_"$run_mode"-$timestamp.log"
+    fi
+
+    touch "$file_path"
+    echo "$file_path"
+}
+
+function storage() { 
+    tee -a "$path"; 
+}
```

Comment:
This isn't likely to be a common function - please move to cmd.sh. Also, I think it would be cleaner to just replace the "echo" calls with something like an "log_echo" function. Also, I'd prefer to see two echo commands in that function instead of using tee, which is going to spawn a new process.

Reply:
That's a good idea. Ill make that change.